### ISSUE TITLE
Can't extract the signature from APK file - bug is still not fixed

### ISSUE THREAD

**Issue by Tobi823** [Tobi823/ffupdater#334]
@PsykoRider

```
java.lang.RuntimeException: Failed to install FIREFOX_NIGHTLY in the foreground.
      at de.marmaro.krt.ffupdater.DownloadActivity.installApp(Unknown Source:171)
      at de.marmaro.krt.ffupdater.DownloadActivity.access$installApp(Unknown Source:0)
      at de.marmaro.krt.ffupdater.DownloadActivity$installApp$1.invokeSuspend(Unknown Source:11)
      at w3.a.resumeWith(Unknown Source:8)
      at i4.l0.run(Unknown Source:102)
      at android.os.Handler.handleCallback(Handler.java:938)
      at android.os.Handler.dispatchMessage(Handler.java:99)
      at android.os.Looper.loopOnce(Looper.java:210)
      at android.os.Looper.loop(Looper.java:299)
      at android.app.ActivityThread.main(ActivityThread.java:8293)
      at java.lang.reflect.Method.invoke(Native Method)
      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1045)
Caused by: de.marmaro.krt.ffupdater.installer.exception.InstallationFailedException: Can't validate the signature of the APK file.
      at de.marmaro.krt.ffupdater.installer.impl.AbstractAppInstaller.install2Internal(Unknown Source:330)
      at de.marmaro.krt.ffupdater.installer.impl.AbstractAppInstaller.access$install2Internal(Unknown Source:0)
      at de.marmaro.krt.ffupdater.installer.impl.AbstractAppInstaller$startInstallation$2.invokeSuspend(Unknown Source:32)
      at w3.a.resumeWith(Unknown Source:8)
      at i4.l0.run(Unknown Source:102)
      at kotlinx.coroutines.internal.e.run(Unknown Source:11)
      at kotlinx.coroutines.scheduling.j.run(Unknown Source:2)
      at kotlinx.coroutines.scheduling.a$a.run(SourceFile:79)
Caused by: java.lang.IllegalArgumentException: Can't extract the signature from APK file '/storage/emulated/0/Android/data/de.marmaro.krt.ffupdater/files/Download/org_mozilla_fenix_2023-02-03 17:06.apk', length: 27975860, absolutePath: /storage/emulated/0/Android/data/de.marmaro.krt.ffupdater/files/Download/org_mozilla_fenix_2023-02-03 17:06.apk, isFile: true, execute: false, read: true, write: false
      at de.marmaro.krt.ffupdater.security.PackageManagerUtil.getPackageArchiveInfo(SourceFile:153)
      at de.marmaro.krt.ffupdater.security.FingerprintValidator$checkApkFile$2.invokeSuspend(Unknown Source:48)
      at de.marmaro.krt.ffupdater.security.FingerprintValidator$checkApkFile$2.invoke(Unknown Source:8)
      at de.marmaro.krt.ffupdater.security.FingerprintValidator$checkApkFile$2.invoke(Unknown Source:4)
      at i4.d0.y(SourceFile:5)
      at a2.i.k0(SourceFile:86)
      at de.marmaro.krt.ffupdater.security.FingerprintValidator.checkApkFile(SourceFile:9)
      at de.marmaro.krt.ffupdater.installer.impl.AbstractAppInstaller.install2Internal(Unknown Source:142)
      ... 7 more
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.2 (128) release |
| Device | POCO F2 Pro (lmi_eea, lmi, lmi) |
| Manufacturer | POCO (Xiaomi) |
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 12 (SDK: 31) |
| OS | non-pangu-pod-xztq7, builder, release-keys, 1672051495000 |

**Comment by Tobi823** [comment]
I think that FFUpdater tried to install this version https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.nightly.2023.02.03.revision.8cbc47b3f4a537fe44f40314727302fe590d2893/arm64-v8a

**Comment by Tobi823** [comment]
But the file that your smartphone has downloaded and tried to install is only 27_975_860 bytes (27 MB) big. But the real APK files should be 91_276_668 bytes (91 MB) for arm64 / 85_378_458 bytes (85 MB). So for som reason the wrong file was downloaded (or maybe not fully downloaded?)

**Comment by Tobi823** [comment]
I suspect that the download server temporarily offers the wrong or a incomplete file for download

**Comment by Tobi823** [comment]
I released version 77.7.3 which includes a download size check. If the downloaded file is too big or too small, then the app installation/update will be aborted.
